pallets/Polkadex_Fungible_Assets/src/lib.rs

```diff
@@ -0,0 +1,320 @@
+            /// Who can use this function? :
+            /// From where balace is coming or is it minting
+            /// Private Token :- Give
+            ensure!(!<InfoVesting<T>>::contains_key(&account), <Error<T>>::VestingInfoExists);
+            let current_block_no = <system::Module<T>>::block_number();
+            let vesting_info = VestingInfo::from(amount, rate, current_block_no);
+            <InfoVesting<T>>::insert(account, vesting_info);
+            Ok(())
+        }
+
+        /// Set Metadata
+        #[weight = 10000]
+        pub fn set_metadata_fungible(origin, asset_id: T::CurrencyId, metadata: AssetMetadata) -> DispatchResult {
+            let who: T::AccountId = ensure_signed(origin)?;
+            ensure!(<InfoAsset<T>>::contains_key(&asset_id), <Error<T>>::AssetIdNotExists);
+            let creator: AssetInfo<T> = Self::get_assetinfo(asset_id);
+            ensure!(who == creator.creator, <Error<T>>::NotTheOwner);
+            InfoAsset::<T>::try_mutate(&asset_id, |ref mut asset_info| {
+                asset_info.metadata = Some(metadata);
+                Self::deposit_event(RawEvent::MetadataAdded(asset_id, who));
+                Ok(())
+            })
+        }
+
+        /// Minting
+        #[weight = 10000]
+        pub fn mint_fungible(origin,to: T::AccountId, asset_id: T::CurrencyId, amount: T::Balance) -> DispatchResult {
+            let who: T::AccountId = ensure_signed(origin)?;
+            Self::mint_token(&who, &to,asset_id, amount)?;
+            Self::deposit_event(RawEvent::AmountMinted(asset_id, who, amount));
+            Ok(())
+        }
+
+        /// Burn
+        #[weight = 10000]
+        pub fn burn_fungible(origin, asset_id: T::CurrencyId, amount: T::Balance) -> DispatchResult {
+            let who: T::AccountId = ensure_signed(origin)?;
+            Self::burn_token(&who,asset_id, amount)?;
+            Self::deposit_event(RawEvent::AmountBurnt(asset_id, who, amount));
+            Ok(())
+        }
+
+        /// Attest Token
+        #[weight = 10000]
+        pub fn attest_token(origin, asset_id: T::CurrencyId) -> DispatchResult {
+            T::GovernanceOrigin::ensure_origin(origin)?;
+            ensure!(<InfoAsset<T>>::contains_key(&asset_id), <Error<T>>::AssetIdNotExists);
+            InfoAsset::<T>::try_mutate(&asset_id, |ref mut asset_info| {
+                asset_info.is_verified = true;
+                Self::deposit_event(RawEvent::TokenVerified(asset_id));
+                Ok(())
+            })
+        }
+
+        /// Modify Token Registration
+        #[weight = 10000]
+        pub fn modify_token_deposit_amount(origin, pdx_amount: T::Balance) -> DispatchResult {
+            T::GovernanceOrigin::ensure_origin(origin)?;
+            <FixedPDXAmount<T>>::put::<T::Balance>(pdx_amount);
+            Self::deposit_event(RawEvent::TokenDepositModified(pdx_amount));
+            Ok(())
+        }
+
```

Comment:
We need one more dispatchable functions for claiming the vested tokens

fn claim( vestingID): something like this, We need to identify each vesting using some identifiers